### PR TITLE
Jp/#1617 energy calc updates

### LINKED ISSUE #1617: Remove energyCalc Param and Use Default
Please remove param enableNewVehicleEnergyConsumptionLogic from configs. Instead let's just let the presence of a file path in vehicleTypesFile, columns (primaryVehicleEnergyFile, secondaryVehicleEnergyFile) dictate whether the more complicate energy calc is performed.

Then, we need to debug why no energy calc is performed for vehicles that don't have an energy file defined. I.e. the fallback is not being used.

### PR DESCRIPTION
**Pull request by JustinPihony**
Closes #1617

---
This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/1626)

### CODE REVIEW

**Comment by JustinPihony** [comment]
test!

**Comment by JustinPihony** [comment]
test!

**Comment by REASY** [comment]
test!

**Comment by JustinPihony** [comment]
test!

**Comment by JustinPihony** [comment]
test!

**Comment by JustinPihony** [comment]
The build passed, and this is only waiting for approval from somebody - @colinsheppard ?

**Comment by JustinPihony** [comment]
test!

**Comment by JustinPihony** [comment]
test!

**Comment by JustinPihony** [comment]
test!

**Comment by JustinPihony** [comment]
test!

**Comment by JustinPihony** [comment]
test!

File: src/main/scala/beam/agentsim/agents/vehicles/VehicleEnergy.scala [hunk 1]
```diff
@@ -210,33 +223,35 @@ class VehicleEnergy(
...
-      .getOrElse(fallBack(fuelConsumptionData))
-    /*.map(x=>(x,Option(gradePercent)))
+    if (!vehicleEnergyMappingExistsFor(fuelConsumptionData.vehicleType)) { fallBack(fuelConsumptionData) } else {
+      val BeamVehicle.FuelConsumptionData(
+        linkId,
+        vehicleType,
+        numberOfLanesOption,
+        _,
+        _,
+        _,
```
Comment (carloscaldas) [inline src/main/scala/beam/agentsim/agents/vehicles/VehicleEnergy.scala]: I believe is a good idea pass the parameters names whenever parameters >= 4

Comment (JustinPihony) [reply src/main/scala/beam/agentsim/agents/vehicles/VehicleEnergy.scala]: I disagree - the names extracted are the ones I believe should be at focus here.

**Comment by JustinPihony** [comment]
test!

**Pull request merged**